integrationTests/buildSrc/gretty-integrationTest/src/main/groovy/org/akhikhl/gretty/internal/integrationTests/BasePlugin.groovy

```diff
@@ -100,6 +102,12 @@ class BasePlugin implements Plugin<Project> {
       project.rootProject.task('wrapper', type: Wrapper) {
         gradleVersion = '6.8.3'
       }
+
+    project.tasks.withType(Test).configureEach {
+      if (GradleVersion.current().baseVersion.version.startsWith("7.")) {
+        useJUnitPlatform()
```

Comment:
Can you explain what this is and why it's needed?

Reply:
That's the opt-in to JUnit 5. I think the chain goes as follows.
- Gradle needs Groovy 3.
- Need to upgrade Spock to cope with Groovy 3.
- Analyze exception during test compilation complaining about missing class from JUnit 4.
- Realize that [Spock has migrated to JUnit 5](https://spockframework.org/spock/docs/2.0-M2/release_notes.html#_2_0_m1_2019_12_31).
-  Realize that the "test run" of the Gradle 7 job is hollow, it is just turning of servlet containers on and off dozens of times without actually executing any of the test methods due to failed test discovery 🤣 

So on Gradle 7, turning on JUnit 5 support in the test task causes Gradle to use JUnit 5 ("the platform") as test runner, and to recognize all Spock tests again.
So for Gradle 6, given old Spock, we still use JUnit 4 as test runner. We could try to unify, and chain following building blocks together: "JUnit 5 as test runner + adapter for JUnit 4 tests (aka "vintage engine") + old Spock". That would get rid of the Gradle version switch above. I'll give it a shot soon.

Reply:
I see, thanks for the explanation. No need to spend your time (unless you want to :) ) on that, I just wanted to understand what this exactly does. Thanks!